app/agents_v2/primary_agent/feedme_knowledge_tool.py

```
@@ -0,0 +1,468 @@
+        except Exception as e:
+            logger.error(f"Failed to initialize FeedMe connector: {e}")
+            _feedme_connector = None
+    return _feedme_connector
+
+
+@tool("enhanced-mailbird-kb-search", args_schema=EnhancedKBSearchInput)
+def enhanced_mailbird_kb_search(
+    query: str,
+    context: Optional[Dict[str, Any]] = None,
+    max_results: int = 5,
+    search_sources: List[str] = ["knowledge_base", "feedme"],
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)